components/style/values/specified/mod.rs

```diff
@@ -655,21 +654,15 @@ impl Shadow {
             if !lengths_parsed {
                 if let Ok(value) = input.try(|i| Length::parse(context, i)) {
                     lengths[0] = value;
-                    let mut length_parsed_count = 1;
-                    while length_parsed_count < length_count {
-                        if let Ok(value) = input.try(|i| Length::parse(context, i)) {
-                            lengths[length_parsed_count] = value
-                        } else {
-                            break
+                    lengths[1] = try!(Length::parse(context, input));
+                    if let Ok(value) = input.try(|i| Length::parse_non_negative(i)) {
+                        lengths[2] = value;
+                        if !disable_spread_and_inset {
```

Comment:
That's right that you should not attempt to parse the spread radius if the option is set. I'm not asking you to remove the check here, but the one at the end of the function. GitHub doesn't allow me to comment on unchanged lines, so I add comment at this line. Sorry if that made you confused.